### PR TITLE
board: add support for seeed xiao esp32s3

### PR DESCRIPTION
**Pull request by LynnL4**
## Checklist

- [x] Please provide specific title of the PR describing the change
- [x] If you are adding an new boards, please make sure
  - [x] Provide link to your allocated VID/PID if applicable
  - [x] Add your board to [action ci](/.github/workflows) in correct workflow and alphabet order for release binary
  - [x] `UF2_BOARD_ID` in your board.h follow correct format from [uf2 specs](https://github.com/microsoft/uf2#files-exposed-by-bootloaders)

### CODE REVIEW

File: ports/espressif/boards/seeed_xiao_esp32s3/board.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+#define LED_PIN               21
+#define LED_STATE_ON          0
+
+//--------------------------------------------------------------------+
+// USB UF2
+//--------------------------------------------------------------------+
+
+#define USB_VID           0x2886
+#define USB_PID           0x0056
```
Comment (hathach) [inline ports/espressif/boards/seeed_xiao_esp32s3/board.h:49]: please provide link to the allocated vid/pid

**Review by hathach — CHANGES_REQUESTED** [review]
thanks for your PR, please address question in the reviews

File: ports/espressif/boards/seeed_xiao_esp32s3/board.h (hunk 1 above)
Comment (LynnL4) [reply ports/espressif/boards/seeed_xiao_esp32s3/board.h:49]: hi, seeed has own VID 0x2886
https://devicehunt.com/search/type/usb/vendor/2886/device/any

Comment (hathach) [reply ports/espressif/boards/seeed_xiao_esp32s3/board.h:49]: I see, but did they allocate the vid 0x0056 for this board as tinyuf2 bootloader ?

Comment (LynnL4) [reply ports/espressif/boards/seeed_xiao_esp32s3/board.h]: The PID is assigned internally. As usual, I have configured the bootloader PID as 0x80XX and the APP PID as 0x00XX.

**Review by hathach — APPROVED** [review]
perfect, thank you.

File: ports/espressif/boards/seeed_xiao_esp32s3/board.h (hunk 1 above)
Comment (hathach) [reply ports/espressif/boards/seeed_xiao_esp32s3/board.h]: ah thanks for clarification, all is good

**Pull request merged**